dir.rb

```diff
@@ -108,114 +194,157 @@ def self.open(name, encoding: nil, &block)
   #
-  #   Note, this will not match Unix-like hidden files (dotfiles).  In order
-  #   to include those in the match results, you must use the
-  #   File::FNM_DOTMATCH flag or something like <code>"{*,.*}"</code>.
+  # Output:
   #
-  # <code>**</code>::
-  #   Matches directories recursively if followed by <code>/</code>.  If
-  #   this path segment contains any other characters, it is the same as the
-  #   usual <code>*</code>.
+  #   config.h
+  #   lib
+  #   main.rb
   #
-  # <code>?</code>::
-  #   Matches any one character. Equivalent to <code>/.{1}/</code> in regexp.
+  # Each pattern string is expanded
+  # according to certain metacharacters;
+  # examples below use the
+  # {Ruby project file tree}[https://github.com/ruby/ruby]:
   #
-  # <code>[set]</code>::
-  #   Matches any one character in +set+.  Behaves exactly like character sets
-  #   in Regexp, including set negation (<code>[^a-z]</code>).
+  # - <tt>'*'</tt>: Matches any substring in an entry name,
+  #   similar in meaning to regexp <tt>/.*/mx</tt>;
+  #   may be restricted by other values in the pattern strings:
   #
-  # <code>{p,q}</code>::
-  #   Matches either literal <code>p</code> or literal <code>q</code>.
-  #   Equivalent to pattern alternation in regexp.
+  #   - <tt>'*'</tt> matches all entry names:
   #
-  #   Matching literals may be more than one character in length.  More than
-  #   two literals may be specified.
+  #       Dir.glob('*').take(3)  # => ["BSDL", "CONTRIBUTING.md", "COPYING"]
   #
-  # <code>\\</code>::
-  #   Escapes the next metacharacter.
+  #   - <tt>'c*'</tt> matches entry names beginning with <tt>'c'</tt>:
   #
-  #   Note that this means you cannot use backslash on windows as part of a
-  #   glob, i.e.  <code>Dir["c:\\foo*"]</code> will not work, use
-  #   <code>Dir["c:/foo*"]</code> instead.
+  #       Dir.glob('c*').take(3) # => ["CONTRIBUTING.md", "COPYING", "COPYING.ja"]
   #
-  # Examples:
+  #   - <tt>'*c'</tt> matches entry names ending with <tt>'c'</tt>:
+  #
+  #       Dir.glob('*c').take(3) # => ["addr2line.c", "array.c", "ast.c"]
+  #
+  #   - <tt>'\*c\*'</tt> matches entry names that contain <tt>'c'</tt>,
+  #     even at the beginning or end:
+  #
+  #       Dir.glob('*c*').take(3) # => ["CONTRIBUTING.md", "COPYING", "COPYING.ja"]
+  #
+  #   Does not match Unix-like hidden entry names ("dot files").
+  #   To include those in the matched entry names,
+  #   use flag IO::FNM_DOTMATCH or something like <tt>'{*,.*}'</tt>.
+  #
+  #  - <tt>'**'</tt>: Matches entry names recursively
+  #    if followed by  the slash character <tt>'/'</tt>:
+  #
+  #      Dir.glob('**/').take(3) # => ["basictest/", "benchmark/", "benchmark/gc/"]
+  #
+  #    If the string pattern contains other characters
+  #    or is not followed by a slash character,
+  #    it is equivalent to <tt>'*'</tt>.
+  #
+  # - <tt>'?'</tt> Matches any single character;
+  #   similar in meaning to regexp <tt>/.{1}/</tt>:
```

Comment:
Doesn't the regex `/./` mean the same as `/.{1}/`?

Reply:
Fixed.